build_runner_core/lib/src/generate/build_definition.dart

```diff
@@ -64,6 +67,122 @@ class BuildDefinition {
       _Loader(environment, options, buildPhases).prepareWorkspace();
 }
 
+/// Understands how to find all assets relevant to a build as well as compute
+/// updates to those assets.
+class AssetTracker {
+  final AssetGraph _assetGraph;
+  final RunnerAssetReader _reader;
+  final TargetGraph _targetGraph;
+
+  AssetTracker(this._assetGraph, this._reader, this._targetGraph);
+
+  /// Checks for and returns any file system changes compared to the current
+  /// state of the asset graph.
+  Future<Map<AssetId, ChangeType>> collectChanges() async {
```

Comment:
Would be nice to get some unit tests in this package as well.

Reply:
done